pep-0587.rst

```diff
@@ -218,24 +239,41 @@ Initialization with PyConfig
 
 The ``PyConfig`` structure contains all parameters to configure Python.
 
-Example of Python initialization enabling the isolated mode::
+Example::
 
-    PyConfig config = PyConfig_INIT;
-    config.isolated = 1;
+    PyInitError err;
+    _PyCoreConfig config = _PyCoreConfig_INIT;
+
+    err = Py_InitializeFromConfig(&config);
+    PyConfig_Clear(&config);
 
-    PyInitError err = Py_InitializeFromConfig(&config);
     if (Py_INIT_FAILED(err)) {
         Py_ExitInitError(err);
     }
-    /* ... use Python API here ... */
-    Py_Finalize();
 
+``PyConfig`` methods:
+
+* ``PyInitError PyConfig_SetString(char **config_str, const char *str)``:
+  Set a config string field from *str* (copy the string)
+* ``PyInitError PyConfig_SetWideString(wchar_t **config_str, const wchar_t *str)``:
+  Set a config wide string field from *str* (copy the string)
+* ``PyInitError PyConfig_SetWideStringFromString(wchar_t **config_str, const char *str)``:
+  Decode *str* using ``Py_DecodeLocale()`` and set the result into ``*config_str``.
+  Pre-initialize Python if needed to ensure that encodings are properly
+  configured.
+* ``PyInitError PyConfig_SetArgv(PyConfig *config, int argc, char **argv)``:
+  Set command line arguments (decode bytes). Pre-initialize Python if
+  needed to ensure that encodings are properly configured.
+* ``PyInitError PyConfig_SetWideArgv(PyConfig *config, int argc, wchar_t **argv)``:
+  Set command line arguments (wide characters).
+* ``PyInitError PyConfig_Read(_PyCoreConfig *config)``:
+  Read all Python configuration
+* ``void PyConfig_Clear(_PyCoreConfig *config)``:
+  Release memory
```

Comment:
I have a lot more thoughts about this part of the API, but I like the direction. Let's chat :)